### PR TITLE
Add unit tests for `WithNewtonsoftJsonHubProtocol`

### PR DESCRIPTION
**Pull request by Y-Sindo**

### CODE REVIEW

File: AzureSignalR.sln [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.28714.193
+# Visual Studio Version 17
```
Comment (vwxyzh) [inline AzureSignalR.sln:2]: revert

File: test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+using Moq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Xunit;
+
+namespace Microsoft.Azure.SignalR.Management.NetCore21.Tests
+{
+    public class AddNewtonsoftFacts
+    {
+        public static IEnumerable<object[]> SerializdObject_SerializerSettings
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs:23]: remove `_`

File: test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            {
+                yield return new object[] { new HubMethodArgument(), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore } };
+                yield return new object[] { new HubMethodArgument { Version = new(1, 1, 1) }, new JsonSerializerSettings { Converters = new JsonConverter[] { new VersionConverter() } } };
+            }
+        }
+
+        #region Persistent
+
+        [Fact]
+        public async Task Test_PersistentMode_DefaultConfiguration()
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs:35]: let's follow the same naming pattern as other tests in our repo

File: test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                .CreateAsync("hubName", default);
+            await serviceHubContext.Clients.All.SendAsync(methodName, message);
+
+            var sentMessage = connectionFactory.CreatedConnections.Single().Value.SelectMany(conn => ((TestServiceConnection)conn).ReceivedMessages).Single() as BroadcastDataMessage;
+            var protocol = new JsonHubProtocol();
+            var exptectedPayload = protocol.GetMessageBytes(new InvocationMessage(methodName, new object[] { message }));
+            var actualPayload = sentMessage.Payloads[protocol.Name];
+            Assert.True(exptectedPayload.Span.SequenceEqual(actualPayload.Span));
+
+            await serviceHubContext.DisposeAsync();
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs:63]: wrap in `try..finally`?

File: test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs (hunk 2 above)
Comment (vicancy) [reply test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs:23]: we usually only use a-z in names

File: test/Microsoft.Azure.SignalR.Management.Tests/Serialization/AddNewtonsoftFacts.cs [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+                    o.ConnectionString = FakeEndpointUtils.GetFakeConnectionString(1).Single();
+                    o.ServiceTransportType = ServiceTransportType.Persistent;
+                })
+                .WithRouter(routerMock.Object)
+                .ConfigureServices(services => services.AddSingleton<IServiceConnectionFactory>(connectionFactoryAlias));
+        }
+
+        #endregion Persistent
+
+        public class HubMethodArgument
```
Comment (vicancy) [inline test/Microsoft.Azure.SignalR.Management.Tests/Serialization/AddNewtonsoftFacts.cs:197]: we only have `private sealed class` as inner classes

Comment (Y-Sindo) [reply test/Microsoft.Azure.SignalR.Management.Tests/Serialization/AddNewtonsoftFacts.cs:197]: `HubMethodArgument` is used in `[MemberData(nameof(SerializdObject_SerializerSettings))]`  and `MemberData` must be public

File: test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        #region Persistent
+
+        [Fact]
+        public async Task TestPersistentModeWithSerializationBehaviour()
+        {
+            ServiceHubContextImpl serviceHubContext = null;
+            try
+            {
+                serviceHubContext = await CreatePersistentServiceHubContextBuilder(out _)
+                .CreateAsync("hubName", default) as ServiceHubContextImpl;
```
Comment (vwxyzh) [inline test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs:41]: indent

File: test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                var hubProtocol = serviceHubContext.ServiceProvider.GetRequiredService<IHubProtocol>();
+
+                //default is Newtonsoft
```
Comment (vwxyzh) [inline test/Microsoft.Azure.SignalR.Management.NetCore21.Tests/Serialization/AddNewtonsoftFacts.cs:44]: Assert?

File: AzureSignalR.sln [hunk 8]
```diff
@@ -73,10 +73,12 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "emulator", "emulator", "{AD
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.SignalR.Emulator", "src\Microsoft.Azure.SignalR.Emulator\Microsoft.Azure.SignalR.Emulator.csproj", "{ABAAAEA8-40B3-4178-A7FD-09779A5DC757}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ChatSample.Net50", "samples\ChatSample\ChatSample.Net50\ChatSample.Net50.csproj", "{29EF376E-DF94-419F-9DF2-50756B60DF38}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ChatSample.Net50", "samples\ChatSample\ChatSample.Net50\ChatSample.Net50.csproj", "{29EF376E-DF94-419F-9DF2-50756B60DF38}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.SignalR.Emulator.Tests", "test\Microsoft.Azure.SignalR.Emulator.Tests\Microsoft.Azure.SignalR.Emulator.Tests.csproj", "{870A10E3-D17A-4239-9715-5610FFE1FC76}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Azure.SignalR.Management.NetCore21.Tests", "test\Microsoft.Azure.SignalR.Management.NetCore21.Tests\Microsoft.Azure.SignalR.Management.NetCore21.Tests.csproj", "{5D557F79-88A0-429F-89C2-63BFB8260E02}"
```
Comment (vicancy) [inline AzureSignalR.sln:80]: just realized that we don't need another project, right? one project can support multiple targetFrameworks

Comment (Y-Sindo) [reply AzureSignalR.sln:80]: But in the existing `Management.Test` project, there is many expressions not supported by .Net Core 2.1 and need refactoring. I think add a new project would be cleaner.

Comment (vicancy) [reply AzureSignalR.sln:80]: sounds acceptable

**Comment by Y-Sindo** [comment]
We have refactored the implementation to use `ObjectSerializer` later which works on all target frameworks. No need to have different test projects for different tfm.

**Pull request closed without merging**